packages/react-components/react-swatch-picker-preview/docs/Spec.md

```diff
@@ -0,0 +1,290 @@
+
+- `square` (default)
+- `circular`
+- `rounded`
+
+Border radius for rounded shape can be customized via CSS.
+
+### Size
+
+- `extraSmall`: 20px
+- `small`: 24px
+- `medium` (default): 28px
+- `large`: 32px
+
+Custom size can be set by overriding `width` and `height` of the ColorSwatch or ImageSwatch.
+
+### Density/Gap
+
+Gap is 2px by default.
+Horizontal and vertical density can be changed via CSS.
+
+### States
+
+- `hover`
+- `selected`
+- `focused`
+- `pressed`
+
+## API
+
+## Structure
+
+### Components
+
+| Component    | Purpose                                                |
+| ------------ | ------------------------------------------------------ |
+| SwatchPicker | Group which can represent swatches as a row or a grid. |
+| ColorSwatch  | Swatch for a color                                     |
+| ImageSwatch  | Swatch for an image, texture or a pattern              |
+
+## SwatchPicker component
+
+#### Anatomy
+
+![visual anatomy of the SwatchPicker component](./assets/todo.png)
+
+#### DOM
+
+```HTML
+<div role="radiogroup" aria-labelledby="colors">
+  {children}
+</div>
+```
+
+#### ColorPicker structure
+
+- root `div` element
+
+#### API
+
+| Property | Values                                   | Default  | Purpose                         |
+| -------- | ---------------------------------------- | -------- | ------------------------------- |
+| shape    | `square`, `circular`, `rounded`          | `square` | Sets shape                      |
+| size     | `extraSmall`, `small`, `medium`, `large` | `medium` | Defines size of the Swatch cell |
+| id       | number                                   |          | Sets swatch id                  |
+| layout   | `row`, `grid`                            | `grid`   | Sets layout of the SwatchPicker |
+| onChange |                                          |          |                                 |
+
+## ColorSwatch component
+
+is used for picking colors:
+
+- solid color
+- gradient
+
+#### Anatomy
+
+![visual anatomy of the SwatchPicker component](./assets/todo.png)
+
+#### DOM
+
+```HTML
+<span name="color" value="{color-id}" style="background: red;">
+  <input type="radio" id="{color-id}" name="color" value="red">
+</span>
+<span name="color" value="{color-id}" style="background: rgb(189, 255, 104);">
+  <input type="radio" id="{color-id}" name="color" value="{color-id}">
+</span>
+```
+
+#### ColorSwatch structure
+
+- root `span` element
+- `input` element
+
+#### API
+
+| Property | Values                                   | Default  | Purpose                         |
+| -------- | ---------------------------------------- | -------- | ------------------------------- |
+| shape    | `square`, `circular`, `rounded`          | `square` | Sets shape                      |
+| size     | `extraSmall`, `small`, `medium`, `large` | `medium` | Defines size of the Swatch cell |
+| id       | number                                   |          | Sets swatch id                  |
```

Comment:
`id` is a string, why it should forced to be a number?

Reply:
I'm more confused as to why it's a required prop? there's no description as to what the prop does

Reply:
I think that it might be needed in cases if the swatch image is different from the final image. For example, in case if final image is too big or swatch has another pattern for better representation. In this case it would be good to have `swatchId`. But this is opened question for discussion. Maybe we won't have such edge cases

Reply:
Resolving it, as no IDs will be supported